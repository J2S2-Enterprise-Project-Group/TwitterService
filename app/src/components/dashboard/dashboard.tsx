/**
 * @author Shannon Phu <shannon.phu@sjsu.edu>
 */
import React, { useCallback } from "react";
import { TweetListItem } from '../tweet-list-item/tweet-list-item';
import List from '@material-ui/core/List';
import { Divider } from "@material-ui/core";

export interface Tweets {
	id_str: string,
	text: string,
}
export interface DashboardProps {
	tweets: Array<Tweets>,
	handleTweetDelete: (id: string) => void
}

export const Dashboard: React.FC<DashboardProps> = ({ tweets, handleTweetDelete }): JSX.Element => {
	/** Updated the dashboard for the delele function
	 *  @author Jagruti Mohanty<jagruti.mohanty@sjsu.edu>
	 *  @author  Jocelyn Baduria<jocelyn.baduria@sjsu.edu>
	 */
	const handleDeleteFn = useCallback((id: string) => {
		handleTweetDelete(id);
	}, []);

	return (
		<List>
			{tweets != null && tweets.map((tweet: { [key: string]: any }, index: number) => {
				return <div key={index}><TweetListItem id={tweet['id_str']} text={tweet['text']} deleteFn={handleDeleteFn} /><Divider variant="fullWidth" component="li" /></div>
			})}
		</List>
	);
};
