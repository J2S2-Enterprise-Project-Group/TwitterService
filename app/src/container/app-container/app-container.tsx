/**
 * @author Shiv Kumar Ganesh <shivkumar.ganesh@sjsu.edu> 
 * @author Shannon Phu <shannon.phu@sjsu.edu>
 */
import { Button, Container, Grid, Typography } from "@material-ui/core";
import React, { useEffect, useState } from "react";
import axios from "axios";
import { useCallback } from "react";
import { Dashboard } from "../../components/dashboard/dashboard";
import { HeaderBar } from "../../components/header/header-bar";
import { API_URL } from "../../config/api-config";
import { TwitterTextbox } from "../../components/twitter-textbox/twitter-textbox";

interface IAppContainer {

}

interface Tweets {
	id_str: string,
	text: string
}

export const AppContainer: React.FC<IAppContainer> = () => {

	const [tweets, setTweets] = useState<Tweets[]>([] as any); // List of tweets to be displayed
	const [tweetText, setTweetText] = useState(''); // Text in the textbox that will be tweeted


	/**
	 * This function handles fetching of the tweets.
	 */
	const getListOfTweets = useCallback(async () => {
		await axios.get(API_URL.LIST_TWEETS)
			.then((response) => {
				setTweets(response.data)
			});
	}, [setTweets]);

	/**
	 * This useEffect helps to load the initial tweets for
	 * the user's API Keys.
	 */
	useEffect(() => {
		getListOfTweets();
	}, [getListOfTweets]);


	const handleTestBoxBlur = useCallback((evt: React.ChangeEvent<any>) => {
		setTweetText(evt.target.value);
	}, [setTweetText])

	const handleButtonClick = useCallback(async (evt: React.ChangeEvent<any>) => {
		await axios.post(API_URL.CREATE_TWEET, {
			status: tweetText
		}).then((response) => {
			// See API response here
			getListOfTweets();
			setTweetText('')
		});
	}, [tweetText, getListOfTweets]);

	return (
		<>
			<HeaderBar />
			<Container maxWidth="sm">
				<Grid container spacing={1} style={{ marginTop: '30px' }} >
					<Grid item xs={12} container
						direction="column"
						justify="flex-end"
						alignItems="flex-end">
						<TwitterTextbox onBlur={handleTestBoxBlur} />
						<div style={{ marginTop: '10px' }}>
							<Button variant="contained" color="primary" onClick={handleButtonClick}>
								Tweet
             			 </Button>
						</div>
					</Grid>
					<Grid item xs={10} container
						direction="column"
						justify="flex-start"
						alignItems="flex-start"></Grid>
					<Typography variant="body1">My Tweets</Typography>
					<br />
					<Dashboard tweets={tweets}></Dashboard>
				</Grid>
			</Container>
		</>
	);
}