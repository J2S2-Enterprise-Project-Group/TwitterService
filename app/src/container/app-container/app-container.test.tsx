/**
 * @author Shiv Kumar Ganesh <shivkumar.ganesh@sjsu.edu>
 */
import React from 'react';
import { mount, shallow } from 'enzyme';
import { AppContainer } from './app-container';
import { Button, Typography } from '@material-ui/core';
import { Dashboard, Tweets } from '../../components/dashboard/dashboard';

const title = 'My Tweets';
const buttonText = 'Tweet';
let wrapped = shallow(<AppContainer />);
let wrapperMount = mount(<AppContainer />);
describe('App Container Component', () => {
	/**
	 * @author Jocelyn Baduria <jocelyn.baduria@sjsu.com>
	 */
	it('Check if the component loaded', () => {
		expect(wrapped).toMatchSnapshot();
	});

	/**
		* @author Shiv Kumar Ganesh <shivkumar.ganesh@sjsu.edu>
		*/
	it('renders the Titles children', () => {
		expect(wrapped.find(Typography).text()).toEqual(title);
	});

	/**
	 * @author Jagruti Mohanty <jagruti.mohanty@sjsu.edu> 
	 */
	it('renders the tweet button', () => {
		expect(wrapped.find(Button).text()).toEqual(buttonText);
		expect(wrapped.find(Button)).toHaveLength(1);
	});

	/**
	 * @author Jagruti Mohanty <jagruti.mohanty@sjsu.edu> 
	 */
	it('renders delete Tweet Icon', () => {
		expect(wrapped.find(Dashboard)).toHaveLength(1);
	});
});