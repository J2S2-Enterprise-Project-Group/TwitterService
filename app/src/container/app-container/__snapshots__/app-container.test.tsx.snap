// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container Component Check if the component loaded 1`] = `
<Fragment>
  <Component />
  <WithStyles(ForwardRef(Container))
    maxWidth="sm"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={1}
      style={
        Object {
          "marginTop": "30px",
        }
      }
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="flex-end"
        container={true}
        direction="column"
        item={true}
        justify="flex-end"
        xs={12}
      >
        <Component
          onBlur={[Function]}
        />
        <div
          style={
            Object {
              "marginTop": "10px",
            }
          }
        >
          <WithStyles(ForwardRef(Button))
            color="primary"
            onClick={[Function]}
            variant="contained"
          >
            Tweet
          </WithStyles(ForwardRef(Button))>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        alignItems="flex-start"
        container={true}
        direction="column"
        item={true}
        justify="flex-start"
        xs={10}
      />
      <WithStyles(ForwardRef(Typography))
        variant="body1"
      >
        My Tweets
      </WithStyles(ForwardRef(Typography))>
      <br />
      <Component
        handleTweetDelete={[Function]}
        tweets={Array []}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
